docs/analyzer-rules/AZFW0008.md

```diff
@@ -8,17 +8,19 @@
 
 ## Cause
 
-This rule is triggered when the EventHubs trigger of an Azure Function is invalid.
+This rule is triggered when an Input Binding or Trigger of an Azure Function has an invalid Cardinality.
 
 ## Rule description
 
-EventHubs triggers must correctly declare a compatible "IsBatched" value and parameter input type. For example, for EventHubs triggers where `IsBatched = true`, the input parameter type must be an iterable collection.
+"Cardinality" dictates whether or not an input is batched together or processed individually. It is defined by using the argument "IsBatched" in an Azure Functions Input Binding or Trigger attribute. When IsBatched is true, Cardinality is set to `Many`. When IsBatched is false, Cardinality is set to `One`.
 
-_**Note:**_ The default value of IsBatched is true. If you are not explicitly providing a value, the parameter type should be a collection type.
+All Input Bindings and Triggers must correctly declare a compatible "IsBatched" value and parameter input type. For example, for EventHubs triggers where `IsBatched = true`, the input parameter type must be an iterable collection. Combining `IsBatched = true` with a parameter of `string[]` is valid, but combining `IsBatched = true` with a parameter of `string` is invalid.
+
+_**Note:**_ The default value of IsBatched changes depending on the Input Binding or Trigger type. The default value of `IsBatched` for each type can be found in the dotnet-isolated [extensions libraries](https://github.com/Azure/azure-functions-dotnet-worker/tree/main/extensions). If a value is not explicitly provided in the named arguments of the attribute, the default value will be used.
 
 ## How to fix violations
 
-If you have an EventHubs trigger where `IsBatched = true` (it is true by default), make sure your input parameter type is an iterable collection.
+If you have an EventHubs trigger where `IsBatched = true` (explicitly or by default), make sure your input parameter type is an iterable collection.
```

Comment:
It could also be a blob trigger or blob input

Reply:
Ah forgot to generalize this section thx for catching